src/com/chunkserver/ChunkServer.java

```diff
@@ -56,29 +59,40 @@ public ChunkServer(){
 			counter = 0;
 		}else{
 			long[] cntrs = new long[fs.length];
-			for (int j=0; j < cntrs.length; j++)
+			for (int j=0; j < cntrs.length; j++){
+				if(fs[j].getName().equals(".DS_Store")){
+					continue;
+				}
 				cntrs[j] = Long.valueOf( fs[j].getName() ); 
-			
+			}
 			Arrays.sort(cntrs);
 			counter = cntrs[cntrs.length - 1];
 		}
 
 		// register with master
+		Socket masterSocket = null;
 		try {
-			Socket masterSocket = new Socket(Master.HOST, Master.PORT);
+			masterSocket = new Socket(Master.HOST, Master.PORT);
 			DataOutputStream out = new DataOutputStream(new BufferedOutputStream(masterSocket.getOutputStream()));
-			byte[] host = InetAddress.getLocalHost().getHostAddress().getBytes();
-			out.writeInt(12 + host.length);
+			//byte[] host = InetAddress.getLocalHost().getHostAddress().getBytes();
+			String host = "localhost";
+			out.writeInt(12 + host.length());
```

Comment:
This should be `host.getBytes().length` since you want to send number of bytes, not number of characters.